api/models/rules/rules.py

```diff
@@ -0,0 +1,38 @@
+from dataclasses import dataclass
+
+from api.models.choices.log_event_types import (
+    CoordinatorLogEventType,
+    GroupLogEventType,
+    StudentLogEventType,
+    TeacherLogEventType,
+    TeacherUnder18LogEventType,
+)
+from api.models.choices.statuses import (
+    CoordinatorStatus,
+    GroupStatus,
+    StudentStatus,
+    TeacherStatus,
+    TeacherUnder18Status,
+)
+
+LogEventType = (
+    CoordinatorLogEventType
+    | GroupLogEventType
+    | StudentLogEventType
+    | TeacherLogEventType
+    | TeacherUnder18LogEventType
+)
+
+
+@dataclass(frozen=True)
+class Result:
```

Comment:
I think it should be different classes for different operation results, maybe with same interface/protocol. Now it seems too complex, with some unnecessary fields in some cases.

Reply:
But maybe I didn't get this design idea at all :D